addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -521,11 +521,26 @@ export class WebsitePreview extends Component {
         // If the iframe is currently displaying an XML file, the body does not
         // exist, so we do not replace the iframefallback content.
         // The iframefallback is hidden in test mode
-        if (!this.websiteContext.edition && this.iframe.el.contentDocument.body && this.iframefallback.el) {
-            this.iframefallback.el.contentDocument.body.replaceWith(this.iframe.el.contentDocument.body.cloneNode(true));
-            this.iframefallback.el.classList.remove('d-none');
-            getScrollingElement(this.iframefallback.el.contentDocument).scrollTop = getScrollingElement(this.iframe.el.contentDocument).scrollTop;
-            this._cleanIframeFallback();
+        const websiteDoc = this.iframe.el?.contentDocument;
+        const fallbackDoc = this.iframefallback.el?.contentDocument;
+        if (!this.websiteContext.edition && websiteDoc && fallbackDoc) {
+            if (websiteDoc.head) {
+                fallbackDoc.head
+                    .querySelectorAll("link[rel='stylesheet'], style")
+                    .forEach((el) => el.remove());
+                for (const el of websiteDoc.head.querySelectorAll(
+                    "link[rel='stylesheet'], style"
+                )) {
+                    fallbackDoc.head.appendChild(el.cloneNode(true));
+                }
+            }
+            if (websiteDoc.body) {
+                fallbackDoc.body.replaceWith(websiteDoc.body.cloneNode(true));
```

Comment:
@odoo/rd-security Could we have your approval for this replaceWith ?
Reason: This commit just moves the existing replaceWith.
It is used to copy the content of the body from the iframe where the website is previewed in edit mode into a second iframe, in order to avoid flickering when switching pages.